packages/containers/src/HeaderBar/HeaderBar.sagas.js

```diff
@@ -0,0 +1,51 @@
+import cmf from '@talend/react-cmf';
+import { call, put, takeEvery, take } from 'redux-saga/effects';
+
+import Connected from './HeaderBar.connect';
+import Constants from './HeaderBar.constant';
+
+/**
+ * This saga takes care of fetching authorized products for the HeaderBar
+ * container according to the provided products URL in the action's payload.
+ * @param {Object} action
+ */
+export function* fetchProducts(action) {
+	const { url, lang } = action.payload;
+	const productsUrl = `${url}?lang=${lang}`;
+
+	yield put(Connected.setStateAction({ productsFetchState: Constants.FETCHING_PRODUCTS }));
+
+	const { response, data } = yield call(cmf.sagas.http.get, productsUrl);
+
+	if (!response.ok) {
```

Comment:
you can switch how the error is handled, this to avoid a negative test expressions and a return inside the if statement